docs/api-testing/use-cases/set-variable.md

```diff
@@ -0,0 +1,261 @@
+
+## Mode: Data
+
+Using this mode, the variable will be evaluated (like the [Expression](/api-testing/composer/expressions/) field), therefore the variable type can be everything. The variable type will depend on the object being evaluated.
+In the Data field, you need to enter a single line expression that returns a value.
+
+For example, you can create a new array in this way:
+
+<img src={useBaseUrl('img/api-testing/set-data-array.png')} alt="Set array"/>
+
+```yaml
+- id: set
+  var: product
+  mode: object
+  object: '["Bluetooth Headphones","Long Sleeve Shirt","Baseball Cap"]'
+```
+
+Then, you can iterate over it using the `each` component or you can invoke a specific item using `${product[1]}` where the number inside the square brackets identifies the position of the item you want to reach out, starting from 0.
+
+Now, let's consider the example below and imagine we have the JSON payload stored in the `payload` variable:
+
+```json
+[
+  {
+    "id": 1,
+    "name": "Baseball Cap",
+    "price": 29.99,
+    "category": "1",
+    "description": "This is product 1!",
+    "quantity": 5,
+    "imageURL": "http://image.com",
+    "color": [
+      "blue",
+      "yellow"
+    ],
+    "createdAt": "2021-11-28T21:58:43.461Z",
+    "updatedAt": "2021-11-28T21:58:43.461Z"
+  },
+  {
+    "id": 2,
+    "name": "Long Sleeve Shirt",
+    "price": 39.99,
+    "category": "1",
+    "description": "This is product 2!",
+    "quantity": 7,
+    "imageURL": "http://image.com",
+    "color": [
+      "blue",
+      "yellow",
+      "red"
+    ],
+    "createdAt": "2021-11-28T21:58:43.461Z",
+    "updatedAt": "2021-11-28T21:58:43.461Z"
+  },
+  {
+    "id": 3,
+    "name": "Bluetooth Headphones",
+    "price": 49.99,
+    "category": "1",
+    "description": "This is product 3!",
+    "quantity": 50,
+    "imageURL": "http://image.com",
+    "color": [
+      "green",
+      "yellow"
+    ],
+    "createdAt": "2021-11-28T21:58:43.462Z",
+    "updatedAt": "2021-11-28T21:58:43.462Z"
+  }
+]
+```
+
+If you write the following:
+
+<img src={useBaseUrl('img/api-testing/set-data.png')} alt="Set data"/>
+
+```yaml
+- id: set
+  var: product
+  mode: object
+  object: payload.filter(it=>it.name=='Bluetooth Headphones')
+```
+
+It will return the following object:
+
+```json
+{
+  "id": 3,
+  "name": "Bluetooth Headphones",
+  "price": 49.99,
+  "category": "1",
+  "description": "This is product 3!",
+  "quantity": 50,
+  "imageURL": "http://image.com",
+  "color": [
+    "green",
+    "yellow"
+  ],
+  "createdAt": "2021-11-28T21:58:43.462Z",
+  "updatedAt": "2021-11-28T21:58:43.462Z"
+}
+```
+
+## Mode: Language
+
+This mode is the most advanced way to create your variables. There are two different options available: `Javascript` and `Template`.
+
+#### Lang: Javascript
+
+In this mode you can create your variable by writing a Javascript script in the `Body` field. It can be a complete script with variable declarations or loops.
```

Comment:
```suggestion
In this mode you can create your variable by writing a JavaScript script in the `Body` field. It can be a complete script with variable declarations or loops.
```